### PR TITLE
Added check for psycopg2 python module for nova/tests/reporting

### PR DESCRIPTION
**Pull request by craigcomstock**
Want to check this as early as possible to save on resource usage and people waiting time.

Ticket: ENT-12432
Changelog: none

together:
https://github.com/cfengine/buildscripts/pull/1523
https://github.com/cfengine/nova/pull/2362

### CODE REVIEW

File: build-scripts/build-environment-check [hunk 1]
```diff
@@ -64,6 +64,14 @@ do
     fi
 done
 
+if [ "$PROJECT" = "nova" ]; then
+    . "$BASEDIR"/nova/tests/reporting/find-python.sh # to get PYTHON as the tests do
```
Comment (larsewi) [inline build-scripts/build-environment-check:68]: Can you have space between `.` and path when executing a file? Asking for a friend :sweat_smile:

File: ci/cfengine-build-host-setup.cf [hunk 2]
```diff
@@ -83,9 +83,13 @@ bundle agent cfengine_build_host_setup
       "pkgconfig";
       "perl-IPC-Cmd";
       "perl-devel";
-      "python-psycopg2";
       "xfsprogs";
 
+    (redhat_6|centos_6).(yum_dnf_conf_ok)::
+      "python-psycopg2" comment => "centos-6 provides python2 and psycopg2 for python2 as a package";
+    (redhat_7|centos_7).(yum_dnf_conf_ok)::
+        "python3-psycopg2";
```
Comment (larsewi) [inline ci/cfengine-build-host-setup.cf:91]: Would look nicer if the promisers are align. But I'm not against being creative :wink:  
```suggestion
    (redhat_6|centos_6).(yum_dnf_conf_ok)::
      "python-psycopg2" comment => "centos-6 provides python2 and psycopg2 for python2 as a package";
    (redhat_7|centos_7).(yum_dnf_conf_ok)::
      "python3-psycopg2";
```

**Review by larsewi — APPROVED** [review]

File: build-scripts/build-environment-check (hunk 1 above)
Comment (craigcomstock) [reply build-scripts/build-environment-check:68]: https://www.shellcheck.net/wiki/SC3046, our shebang is /bin/sh so `.` is more proper than `source`.

File: ci/cfengine-build-host-setup.cf (hunk 2 above)
Comment (craigcomstock) [reply ci/cfengine-build-host-setup.cf:91]: yeah.

**Pull request merged**